odbc/src/api/functions.rs

```diff
@@ -2238,6 +2344,70 @@ unsafe fn sql_get_infow_helper(
         add_diag_with_function!(conn_handle, error, "SQLGetInfoW");
     }
     sql_return
+}}
+}
+
+///
+/// [`SQLGetInfoW`]: https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/SQLGetInfo-function
+///
+/// This is the WideChar version of the SQLGetInfo function
+///
+/// # Safety
+/// Because this is a C-interface, this is necessarily unsafe
+///
+#[named]
+#[no_mangle]
+pub unsafe extern "C" fn SQLGetInfoW(
+    connection_handle: HDbc,
+    info_type: USmallInt,
+    info_value_ptr: Pointer,
+    buffer_length: SmallInt,
+    string_length_ptr: *mut SmallInt,
+) -> SqlReturn {
+    panic_safe_exec_clear_diagnostics!(
+        debug,
+        || sql_get_info_helper!(
+            connection_handle,
+            info_type,
+            info_value_ptr,
+            buffer_length,
+            string_length_ptr,
+            i16_len::set_output_wstring_as_bytes,
+            "SQLGetInfoW",
+        ),
+        connection_handle
+    )
+}
+
+///
+/// [`SQLGetInfo`]: https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/SQLGetInfo-function
+///
+///
+/// # Safety
+/// Because this is a C-interface, this is necessarily unsafe
+///
+#[named]
+#[no_mangle]
+pub unsafe extern "C" fn SQLGetInfo(
```

Comment:
PowerBI calls SQLGetInfo instead of SQLGetInfoW for some reason

Reply:
Since always? Their mashup logs very specifically say SQLGetInfo, as well. If I don't put SQLGetInfo in here it simply doesn't work.

Reply:
This is not what I have been seeing.
Here are the old traces, for the beginning of the work with ODBC: https://docs.google.com/document/d/1eMnHvYwhLymsOvepjB7GgM9saN6asMl55Xr33448xes/edit#heading=h.gh965js96g5g
It calls SQLGetInfoW.
How would users be able to user (and Alexi) the latest driver if it was not working and calling SQLGetInfo?

Just a thought here. Last time you were using an old version of PowerBi. Is this one the newest one?

Reply:
What I meant by "doesn't work" is that it wouldn't Direct Query anything that relied on SQLGetInfo output, i.e., any Cast. Should have been more clear there.